### ISSUE TITLE
VRE Data lake

### ISSUE THREAD

**Issue by KoenGreuell** [NaaVRE/NaaVRE#114]
https://github.com/zmzhao/AcademicProjects/issues/212

### Data cubes / datacubes and data lakes: Accessing and storing data from a virtual research environment
NaaVRE (Notebook-as-a-Virtual-Research-Environment) supports data access and storage via any source offering an API. 
However, the current model places the burden on individual virtual lab developers to create ad-hoc data strategies. 
This lack of standardization hinders one of the goals of NaaVRE: 
promoting the reusability of research workflows and enabling effective data sharing and collaboration.

While commercial entities often centralize data using Data Lakes for internal accessibility, 
this model conflicts with the scientific community's aim to make data 
Findable, Accessible, Interoperable, and Reusable (FAIR) across different institutions. 
The scientific state-of-the-art for simplified and standardized data access involves structuring complex, 
multi-dimensional scientific data into Data Cubes (or Hypercubes).

We want to integrate "hot" operational data storage (data actively used within NaaVRE workflows) 
with "cold" external data systems (institutional databases, external repositories, tape storage) 
in a way that inherently promotes and rewards FAIR data principles.

### User stories
As a virtual lab developer,
I want to be able to share data with other virtual lab users,
So that I can effectively collaborate with a team of researchers.

As a virtual lab developer,
I want to rapidly find or develop components to access and store data,
To save development time by eliminating the need to write custom data retrieval code for common institutional or public sources.

As a virtual lab developer,
I want to be supported in gaining insight in what data to store and what data to discard,
To make sound choices when it comes to choosing between recomputing and using storage.

### Current limitations
Retrieving and storing data is not facilitated by NaaVRE, but relies on the virtual lab developer making a strategy to store their data such that the virtual lab is reusable. Additionally, there is no automated system to incentivize or measure a virtual lab's adherence to FAIR data principles regarding its input/output data.

### Drivers
- Virtual labs from the Biodiversity & Macroecology (BIOMAC) lab at UvA:
  - https://github.com/NaaVRE/ALSTreeInd-virtual-lab/issues/1
  - https://github.com/NaaVRE/Laserfarm-virtual-lab/issues/1
  - https://github.com/bmarcos/WiReSIVL/issues/2
- The LTER-LIFE project is looking into creating components to access commonly used data sources.
- Funded project Biodiversity meets Data is looking into data (hyper)cubes.

### Features
- Features for NaaVRE users to simplify sharing hot and cold data.
- Features for NaaVRE to simplify the access to hot and cold data including tape.
- FAIRification Incentive System: Investigate mechanisms (e.g., metadata completeness scoring, data quality checks) to reward and showcase virtual labs or components that adhere to and promote FAIR data practices.
- Features to decide what data to keep and what data to discard.

### Required expertise
- Data (hyper)cubes
- Data lakes
- FAIR
- Incentives in scientific research

### Potential technologies
- https://beacon.maris.nl/
- Registration Data Access Protocol (RDAP) https://cortex.marketplace.pan.dev/marketplace/details/RDAP/

### Related development
- Accessing external databases often requires the use of secrets. Handling secrets in NaaVRE is under development here: https://github.com/NaaVRE/NaaVRE/issues/91 
- Ensuring that data used by a virtual lab (workflow) is FAIR, will be worked on in https://github.com/NaaVRE/NaaVRE/issues/21
- https://github.com/QCDIS/infrastructure/issues/242

### Consider
What is the relationship between data, workflows, source code and users/teams?

**Comment by KoenGreuell** [comment]
https://www.ibm.com/think/topics/data-lake
https://azure.microsoft.com/en-us/resources/cloud-computing-dictionary/what-is-a-data-lake
### Literature questions
- [ ] How do collaboration patterns influence how data is organized? Data may be shared with no one, shared within a team, shared within a company, shared with a consortium, or publicly available. How should these differences be dealt with in the data management strategy?
- [ ] How about different organizations? What is the difference in the data used by a company and a research institute and CERN and how does that influence their data management strategy? 
- [ ] Can you create a flowchart of what you view as relevant examples of data ingestion, data processing, data persistence, data governance, and applications in the context of a company (that exchanges data with other companies), a researcher, a research group / institute and a research community?
- [ ] How to use the full potential of data in a company setting. How to use the full potential of the large amount of distributed data in a research setting?
- [ ] What are existing solutions for data finding and sharing for commercial data analists and scientists?

### Questionaire for domain scientists
Create a questionaire for domain scientists.
- What data do they produce.
- How do they share it.
- What data do they store short term, what data do they store long term?
- What is challenging data to store? Why is there no suitable option for this data?
- How do you find data and find out if it is usable for your project?
Perhaps make a data landscape with some of these scientists?
